hardware/tests/test_detect.py

```diff
@@ -100,6 +104,200 @@ def fake_ethtool_status(arg, arg1):
         detect.get_uuid = fake_get_uuid
         detect_utils.get_lld_status = fake_lld_status
         detect_utils.get_ethtool_status = fake_ethtool_status
+        detect.get_cpus = fake_get_cpus
+
+    def test_get_cpus(self):
```

Comment:
Could you also add a test with fake data from something really primitive, like a VM? Just to make sure we don't require fields that are not really present everywhere.